datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -688,95 +683,77 @@ fn build_batch_from_indices(
         };
         columns.push(array);
     }
-    RecordBatch::try_new(Arc::new(schema.clone()), columns).map(|x| (x, left_indices))
+    RecordBatch::try_new(Arc::new(schema.clone()), columns)
 }
 
+// Get left and right indices which is satisfies the on condition in the Join
```

Comment:
It is easy to forget the `filter` and `equal-conditon` both are `on-condition` and they are combined by `And` implicitly.
```suggestion
// Get left and right indices which is satisfies the on condition (include equal_conditon and filter_in_join) in the Join
```